### ISSUE TITLE
Keyboard-layout independent bindings

### ISSUE THREAD

**Issue by mitinarseny** [riverwm/river#338]
At the moment, `riverctl map` maps keysyms only for given layout. For example, in my config, Mod4+F is to toggle fullscreen. So whenever I switch to `ru` layout, Mod4+F does nothing, since river thinks of it as a different key combination. Is it possible to implement keyboard-layout independent key bingings?

**Comment by ZerdoX-x** [comment]
This also bothers me. Any workaround?

**Comment by pmkap** [comment]
I started to implement this. Just thinking what would be the nicest command interface.

1. `riverctl map-code normal Super 29 spawn foot` (extra command like i3/sway's `bindcode`)
2. `riverctl map -code normal Super 29 spawn foot` (same but no extra command, instead optional arg for `map`)
3. `riverctl map -to-code normal Super Y spawn foot` (like sway's `bindsym --to-code`, would determine the keycode based on the current layout)

I like option 3 because one doesn't have to look up the keycodes, which could be confusing for some users.

**Comment by n1rdeks** [comment]
If u asking me - I prefer -to-code, IMO. Because we need a little standart. Wlroots - Sway. So, River users use --to-code in Sway before trying River...

**Comment by MaxVerevkin** [comment]
Hi, I use a `workman-p` layout and I have numbers swapped with `! @ #` etc. At the moment, neither

```
riverctl map normal Super 1 set-focused-tag 0
```

nor 

```
riverctl map normal Super Exclam set-focused-tag 0
```

function as expected. Would `-to-code` help in this case? (in sway I have `bindsym --to-code $mod+1 workspace number 1`.) If so, can I help somehow?

**Comment by pmkap** [comment]
@MaxVerevkin The second one should work right now. The command is `set-focused-tags` (plural) and this cannot be `0`. So `riverctl map normal Super Exclam set-focused-tags 1` should work just fine.

**Comment by pmkap** [comment]
The problem with `-to-code` is when there are two keyboards. How to decide which keyboard to use for the `-to-code` translation?

Sway doesn't have this problem because mappings are always loaded when sway starts.

**Comment by ifreund** [comment]
Implemented in #577
